src/bidiMapper/modules/cdp/CdpTarget.ts

```diff
@@ -265,7 +270,18 @@ export class CdpTarget {
         handleAuthRequests: stages.auth,
       });
     } else {
-      await this.#cdpClient.sendCommand('Fetch.disable');
+      const blockedRequest = this.#networkStorage
+        .getRequestsByTarget(this)
+        .filter((request) => request.interceptPhase);
+      void Promise.allSettled(
```

Comment:
Pseudo code:
```
if(blockedRequest.isEmpty())
  return await this.#cdpClient.sendCommand('Fetch.disable');
else
  void Promise.allSettled(
Comment
Pseudo code:
```
if(blockedRequest.isEmpty())
  return await this.#cdpClient.sendCommand('Fetch.disable');
else
 void Promise.allSettled(
    blockedRequest.map((request) => request.waitNextPhase)
  ).then(async () => {
          toggleFetchIfNeeded()
});
```